packages/wrangler/src/versions/secrets/index.ts

```diff
@@ -122,18 +122,17 @@ export async function copyWorkerVersionWithNewSecrets({
 	);
 
 	// Filter out secrets because we're gonna inherit them
-	const bindings: WorkerMetadataBinding[] =
-		versionInfo.resources.bindings.filter(
-			(binding) => binding.type !== "secret_text"
-		);
+	let bindings: WorkerMetadataBinding[] = versionInfo.resources.bindings.filter(
+		(binding) => binding.type !== "secret_text"
+	);
 
-	// We cannot upload a DO with a namespace_id so remove it
-	for (const binding of bindings) {
-		if (binding.type === "durable_object_namespace") {
-			// @ts-expect-error - it doesn't exist within wrangler but does in the API
-			delete binding.namespace_id;
-		}
-	}
+	// Inherit all of the existing bindings
+	bindings = bindings.map((binding) => {
+		return {
+			name: binding.name,
+			type: "inherit",
+		};
+	});
 
 	// Add the new secrets
 	for (const secret of secrets) {
```

Comment:
Could simplify inheritance here, we filter out, then map for inherit and then push the filtered out ones to inherit in this for loop